components-java-sdk/src/main/java/io/dapr/components/domain/state/QueryItem.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.domain.state;
+
+import com.google.protobuf.ByteString;
+
+/**
+ * QueryItem is an object representing a single entry in query results.
+ *
+ * @param key The returned item Key.
+ * @param data The returned item Data.
+ * @param etag The returned item ETag.
+ * @param error The returned error string.
+ * @param contentType  The returned content-type.
+ */
+public record QueryItem(String key, ByteString data, String etag, String error, String contentType) {
```

Comment:
I recall that other types also had an overridden constructor that allowed passing `byte[]`.

Reply:
I could add but I didn't because this object is only ever received by the network and presented to the developer as data to be consumed. In normal circumstances, a developer won't create instances of this data type to send it down the wire.

Aside testing, there would be no use-case I could think of of a user ever needed to create this. But now that I mentioned testing it makes sense to add it just in case ;)

Reply:
Ah, yeah, it makes sense that a `byte[]` option might be redundant for request-related types.  Totally your call, then.

Reply:
Added this alt. constructor  in the last commit just in case.